spec/p2p/p2p-top-level.md

```diff
@@ -0,0 +1,310 @@
+*** This is the beginning of an unfinished draft. Don't continue reading! ***
+
+# << Tendermint P2P >>
+
+<!--- > Rough outline of what the component is doing and why. 2-3 paragraphs 
+--->
+
+The Tendermint consists of multiple protocols, namely,
+- Consensus
+- Mempool
+- Evidence
+- Blocksync
+- Statesync
+
+that each plays a role in making sure that validators can produce blocks. These protocols are implemented in so-called reactors (one for each protocol) that encode two functionalities:
+
+- Protocol logic (controlling the local state of the protocols and deciding what messages to send to others, e.g., the rules we find in the arXiv paper)
+
+- Communication. Message exchange with other nodes (Gossip)
+
+Tendermint (as many classic BFT algorithms) have an all-to-all communication pattern (e.g., every validator sends a `precommit` to every other validator). Naive implementations, e.g., maintaining a channel between each of the *N* validators is not scaling to the system sizes of typical Cosmos blockchains (e.g., N = 200 validator nodes + seed nodes + sentry nodes + other full nodes). There is the fundamental necessity to restrict the communication.
```

Comment:
> every validator sends a `precommit` to every other validator

I would rather say it sends the `precommit` to all full nodes (since full nodes also decide)